lib/ansible/module_utils/facts/system/date_time.py

```diff
@@ -52,7 +52,7 @@ def collect(self, module=None, collected_facts=None):
         date_time_facts['iso8601'] = now.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
         date_time_facts['iso8601_basic'] = now.strftime("%Y%m%dT%H%M%S%f")
         date_time_facts['iso8601_basic_short'] = now.strftime("%Y%m%dT%H%M%S")
-        date_time_facts['tz'] = time.strftime("%Z")
+        date_time_facts['tz'] = time.tzname[1 if time.localtime().tm_isdst == 1 else 0]
```

Comment:
We should add a new key in facts stating its usage. This is for backward compatibility and playbooks which are relying on this fact.

Reply:
Also, add an integration test for this change.

Reply:
Like @sivel described here - https://github.com/ansible/ansible/issues/69004#issuecomment-617424338

Reply:
If I correctly understood @sivel 's mentioned comment, he suggested to add a new key(s), because time.tzname returned tuple of non-DST + DST values and the backward compatibility could be broken if we just choose non-DST timezone. I think this patch provides accurate timezone info on a host where ansible gets facts. It should work for all cases and more accurate than `time.strftime("%Z")`

Reply:
We discussed in triage today and there should be a new fact that accounts for the DST state of the system rather than changing the (longstanding) behavior of the existing TZ key.

Reply:
@samdoran Thank you! I've added a new fact `tz_dst` instead.